### PR TITLE
FAD-5698 Subaccount Grants

### PR DESCRIPTION
**Pull request by jonambas**

### CODE REVIEW

File: src/reducers/api-keys.js [hunk 1]
```diff
@@ -30,21 +30,23 @@ export default (state = initialState, { payload, type }) => {
...
     }
 
     case 'LIST_GRANTS_FAIL': {
-      return {
-        ...state,
-        error: payload,
-        grantsLoaded: true,
-        grantsLoading: false
-      };
+      return { ...state, error: payload, grantsLoaded: true, grantsLoading: false };
```
Comment (jonambas) [inline src/reducers/api-keys.js]: Unrelated to changes but - not really sure I like the `force` option within actions. Also feels unnecessary to have both `grantsLoaded` and `grantsLoading`

File: src/selectors/api-keys.js [hunk 2]
```diff
@@ -23,6 +24,10 @@ export const getGrants = createSelector(getGrantsArray, (grants) =>
   _.keyBy(grants, 'key')
 );
 
+export const getSubaccountGrants = createSelector(getSubaccountGrantsArray, (grants) =>
+  _.keyBy(grants, 'key')
+);
+
 export const getLoading = createSelector(
   [getKeysLoading, getGrantsLoading, getSubaccountsLoading],
```
Comment (jonambas) [inline src/selectors/api-keys.js]: Need to add `getSubaccountGrantsLoading` selector to this

File: src/pages/api-keys/DetailsPage.js [hunk 3]
```diff
@@ -33,6 +37,11 @@ export class ApiKeysDetailsPage extends Component {
 
   componentDidMount() {
     this.props.listApiKeys();
```
Comment (jgzamora) [inline src/pages/api-keys/DetailsPage.js]: Does this page need to list keys?

File: src/pages/api-keys/DetailsPage.js [hunk 4]
```diff
@@ -90,7 +99,8 @@ const mapStateToProps = (state, props) => {
     keys,
```
Comment (jgzamora) [inline src/pages/api-keys/DetailsPage.js]: this doesn't look like it's used

File: src/pages/api-keys/DetailsPage.js [hunk 5]
```diff
@@ -90,7 +99,8 @@ const mapStateToProps = (state, props) => {
...
     error,
     grants,
-    loading: getLoading(state)
+    hasSubaccounts: hasSubaccounts(state),
+    loading: getFormLoading(state) || getKeysLoading(state)
```
[the remark this answers is not in the document]
Comment (jonambas) [reply src/pages/api-keys/DetailsPage.js]: The "single" is picked from the list

```
export const getApiKey = createSelector(
  [getApiKeys, getApiKeyId],
  (apiKeys, id) => _.find(apiKeys, { id })
);
```

**Comment by jonambas** [comment]
Added some tests - I think this looks good to go

**Pull request merged**